website/notifications/emails.py

```diff
@@ -9,152 +8,120 @@
 from website.notifications.model import NotificationSubscription
 from website.util import web_url_for
 
-LOCALTIME_FORMAT = '%H:%M on %A, %B %d %Z'
-EMAIL_SUBJECT_MAP = {
-    'comments': '${user.fullname} commented on "${title}".',
-    'comment_replies': '${user.fullname} replied to your comment on "${title}".'
-}
 
-
-def email_transactional(recipient_ids, uid, event, user, node, timestamp, **context):
+def notify(event, user, node, timestamp, **context):
     """
-    :param recipient_ids: mod-odm User object ids
-    :param uid: id of the event owner (Node or User)
-    :param event: name of notification event (e.g. 'comments')
-    :param context: context variables for email template
-        See notify for specifics
-    :return:
+    Retrieves appropriate ***subscription*** and passes user list to store_emails
+    :param event: event that triggered the notification
+    :param user: user who triggered notification
+    :param node: instance of Node
+    :param timestamp: time event happened
+    :param context: optional variables specific to templates
+        target_user: used with comment_replies
+    :return: List of user ids notifications were sent to
     """
-    template = event + '.html.mako'
-    context['title'] = node.title
-    context['user'] = user
-    subject = Template(EMAIL_SUBJECT_MAP[event]).render(**context)
-
-    for user_id in recipient_ids:
-        recipient = website_models.User.load(user_id)
-        email = recipient.username
-        context['localized_timestamp'] = localize_timestamp(timestamp, recipient)
-        message = mails.render_message(template, **context)
-
-        if user._id != recipient._id:
-            mails.send_mail(
-                to_addr=email,
-                mail=mails.TRANSACTIONAL,
-                mimetype='html',
-                name=recipient.fullname,
-                node_id=node._id,
-                node_title=node.title,
-                subject=subject,
-                message=message,
-                url=get_settings_url(uid, recipient)
-            )
-
-
-def email_digest(recipient_ids, uid, event, user, node, timestamp, **context):
-    """ Render the email message from context vars and store in the
-        NotificationDigest objects created for each subscribed user.
+    event_type = "_".join(event.split("_")[-2:])  # tests indicate that this should work with no underscores
```

Comment:
```python

def find_subscription_type(subscription):
    subs_available = list(constants.USER_SUBSCRIPTIONS_AVAILABLE.keys())
    subs_available.extend(list(constants.NODE_SUBSCRIPTIONS_AVAILABLE.keys()))
    for available in subs_available:
        if available in subscription:
            return available
```

Not my most favorite, but it gets all the cases right now without problems.

The issue is if in the future a subscription is added that is a subset of an existing one and it gets to go first, then it will be returned rather than the correct subscription type.